k8s/upgrade/src/plugin/upgrade.rs

```diff
@@ -220,8 +276,26 @@ impl UpgradeEventClient {
         self.upgrade_event.clone()
     }
 
+    /// Delete upgrade events.
+    pub async fn delete_upgrade_events(&self, release_name: String) -> error::Result<()> {
+        let selector = upgrade_event_selector(release_name.as_str(), UPGRADE_JOB_NAME_SUFFIX);
+        let event_lp = ListParams {
+            field_selector: Some(selector.clone()),
+            ..Default::default()
+        };
+        self.api_client()
+            .delete_collection(&DeleteParams::default(), &event_lp)
+            .await
+            .context(error::DeleteEventsWithFieldSelector { field: selector })?;
+        Ok(())
+    }
+
     /// Fetch upgrade events and print.
-    pub async fn get_and_log(&self, release_name: String) -> error::Result<()> {
+    pub async fn get_upgrade_events(
```

Comment:
this function name suggest plural events, but seems we return only one event?

Reply:
renamed it to `get_latest_upgrade_event`